### DISCUSSION TITLE
issue: Errors are not updated for a field that was not touched

### DISCUSSION THREAD

**Discussion by radoslavkarlik** [react-hook-form/react-hook-form#9465] [Q&A]
### Version Number

7.39.7

### Codesandbox/Expo snack

https://codesandbox.io/s/nervous-lucy-0y1zkg

### Steps to reproduce

1. Click on submit to enter the validation mode. Form becomes invalid (checked + empty input)
2. Uncheck the checkbox.

Once you click the submit again then it correctly considers the form as valid.
The resolver is called, just the errors are not updated in the render.
Might be relevant to https://github.com/react-hook-form/react-hook-form/discussions/8820

Alternatively, if you type anything in the input, then it immediately becomes valid.

### Expected behaviour

The form should become valid but it is considered invalid.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
for dependant  validation use trigger or `deps` props from `register`, hook form update error by field level
